### PR TITLE
Make the number of concurrent goroutines in decomposedfs configurable

### PR DESCRIPTION
**Pull request by aduffeck**
The number of goroutines used for listing directories in decomposedfs is now limited to configurable number.

### CODE REVIEW

**Comment by butonic** [comment]
@aduffeck shouldn't we add the config to the storage-system as well?

File: services/storage-users/pkg/config/config.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ type OCISDriver struct {
 	ShareFolder             string `yaml:"share_folder" env:"STORAGE_USERS_OCIS_SHARE_FOLDER" desc:"Name of the folder jailing all shares."`
 	MaxAcquireLockCycles    int    `yaml:"max_acquire_lock_cycles" env:"STORAGE_USERS_OCIS_MAX_ACQUIRE_LOCK_CYCLES" desc:"When trying to lock files, ocis will try this amount of times to acquire the lock before failing. After each try it will wait for an increasing amount of time. Values of 0 or below will be ignored and the default value of 20 will be used."`
 	LockCycleDurationFactor int    `yaml:"lock_cycle_duration_factor" env:"STORAGE_USERS_OCIS_LOCK_CYCLE_DURATION_FACTOR" desc:"When trying to lock files, ocis will multiply the cycle with this factor and use it as a millisecond timeout. Values of 0 or below will be ignored and the default value of 30 will be used."`
+	MaxConcurrency          int    `yaml:"max_concurrency" env:"STORAGE_USERS_OCIS_MAX_CONCURRENCY" desc:"Maximum number of concurrent goroutines. Higher values can cause more load on the system but potentially get work done faster. Values of 0 or below will be ignored and the default value of 100 will be used."`
```
Comment (mmattel) [inline services/storage-users/pkg/config/config.go:116]:
```suggestion
	MaxConcurrency          int    `yaml:"max_concurrency" env:"STORAGE_USERS_OCIS_MAX_CONCURRENCY" desc:"Maximum number of concurrent go-routines. Higher values can potentially get work done faster but will also cause more load on the system. Values of 0 or below will be ignored and the default value of 100 will be used."`
```
Say the positive first 😄 
`Maximum number of concurrent go-routines` ... for what?

File: services/storage-users/pkg/config/config.go [hunk 2]
```diff
@@ -139,6 +140,7 @@ type S3NGDriver struct {
 	ShareFolder             string `yaml:"share_folder" env:"STORAGE_USERS_S3NG_SHARE_FOLDER" desc:"Name of the folder jailing all shares."`
 	MaxAcquireLockCycles    int    `yaml:"max_acquire_lock_cycles" env:"STORAGE_USERS_S3NG_MAX_ACQUIRE_LOCK_CYCLES" desc:"When trying to lock files, ocis will try this amount of times to acquire the lock before failing. After each try it will wait for an increasing amount of time. Values of 0 or below will be ignored and the default value of 20 will be used."`
 	LockCycleDurationFactor int    `yaml:"lock_cycle_duration_factor" env:"STORAGE_USERS_S3NG_LOCK_CYCLE_DURATION_FACTOR" desc:"When trying to lock files, ocis will multiply the cycle with this factor and use it as a millisecond timeout. Values of 0 or below will be ignored and the default value of 30 will be used."`
+	MaxConcurrency          int    `yaml:"max_concurrency" env:"STORAGE_USERS_S3NG_MAX_CONCURRENCY" desc:"Maximum number of concurrent goroutines. Higher values can cause more load on the system but potentially get work done faster. Values of 0 or below will be ignored and the default value of 100 will be used."`
```
Comment (mmattel) [inline services/storage-users/pkg/config/config.go:143]:
```suggestion
	MaxConcurrency          int    `yaml:"max_concurrency" env:"STORAGE_USERS_S3NG_MAX_CONCURRENCY" desc:"Maximum number of concurrent go-routines. Higher values can potentially get work done faster but will also cause more load on the system. Values of 0 or below will be ignored and the default value of 100 will be used."`
```
Say the positive first 😄 
`Maximum number of concurrent go-routines` ... for what?

**Review by butonic — APPROVED** [review]

**Comment by ownclouders** [comment]
E2E tests failed: https://drone.owncloud.com/owncloud/ocis/20771/66/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20771/tracing/space-members-can-be-invited-with-an-expiration-date-alice-2023-3-23-08-02-28.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20771/tracing/space-members-can-be-invited-with-an-expiration-date-brian-2023-3-23-08-02-33.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/ocis/20771/tracing/unstructured-collection-of-testable-space-interactions-alice-2023-3-23-08-04-01.zip 


</p></details>

**Pull request merged**